Make chunk size of database storage driver configurable

Namoshek opened the pull request:
The PR introduces a new `chunk_size` configuration parameter for the `database` storage driver in order to fix the issue explained in #614.

In case someone created a custom version of the `TelescopeServiceProvider` which does not pass the new `$chunkSize` parameter to the database repository, the repository will still work due to the old default being used. Also users relying on the default service provider will have no issues as the old default is used if no configured chunk size is being found.

What I'm not sure about is how to properly document the setting. Personally, I'd prefer if there was a thorough explanation in `config/telescope.php`. There isn't really a common way to document nested configuration options though. Alternatively, we can also add it to the official documentation. Basically, I'm open to suggestions.

delta1186 commented:
**Question**: @taylorotwell when will this merge with be rolled up in a new release?  Thanks, Kevin

The rest are observations from my experiences with trying to get around this issue today and a possible bug or user error. :)

Ironically I ran into this issue today and after a few hours of messing with various telescope config options for ignoring a custom console command, which I wasn't able to get to work.  

protected $signature = '**connect:updateAdLists**';  this is from my custom console command class

I was trying to add connect:updateAdLists to either of the following areas.  Used lower case and how it is in the $signature var.    

The console command just dispatches a job so I also tried to set the following env vars, but I still got the following error as reported and fixed in this PR.  

TELESCOPE_COMMAND_WATCHER=false
TELESCOPE_JOB_WATCHER=false

SQLSTATE[IMSSP]: Tried to bind parameter number 2101.  SQL Server supports a maximum of 2100 parameters. (SQL: insert into [telescope_entries]....

```
'ignore_commands' => [
        'connect:updateAdLists'
    ],

Watchers\CommandWatcher::class => [
            'enabled' => env('TELESCOPE_COMMAND_WATCHER', true),
            'ignore' => ['connect:updateAdLists'],
        ],
```
So I wanted to mention the above in case something is not working as expected in telescope.  Hopefully I am just missing something with the ignore options, but I don't think I am especially when I am supposedly turning the watchers off by setting those env vars to false and telescope still errors because it tries record those entries.

Namoshek commented:
You can always pull in a specific commit through composer by specifying the branch and commit instead of a version:
```json
"require": {
    "laravel/telescope": "2.0#1749dae1"
}
```
This will pull branch `2.0` at commit `1749dae1` (most recent).

The recent changes since the last version can be found here: [v2.0.4...2.0](https://github.com/laravel/telescope/compare/v2.0.4...2.0).

FrittenKeeZ commented:
> You can always pull in a specific commit through composer by specifying the branch and commit instead of a version:
> 
> ```json
> "require": {
>     "laravel/telescope": "2.0#1749dae1"
> }
> ```
> 
> This will pull branch `2.0` at commit `1749dae1` (most recent).
> 
> The recent changes since the last version can be found here: [v2.0.4...2.0](https://github.com/laravel/telescope/compare/v2.0.4...2.0).

You need to add `dev-` for this to work:
```json
"require": {
    "laravel/telescope": "dev-2.0#1749dae1"
}
```
